index.html

```diff
@@ -1,45 +1,451 @@
+            selected by the user. In case not all the
+            <a data-cite="screen-capture#dfn-display-surface">display surfaces</a>
+            selected by the user can be captured, {{MediaDevices/getDisplayMediaSet}} must return
+            a promise <a>rejected</a> with a new {{InvalidStateError}}.
+          </p>
+          <p>
+            The <a href="#dfn-user-agent">user agent</a> must apply {{DisplayMediaStreamOptions.video}}
+            constraints to to each of the selected
+            <a data-cite="screen-capture#dfn-display-surface">display surfaces</a> in the same way as it does for
+            {{MediaDevices/getDisplayMedia()}}.
+          </p>
+          <p>
+            When prompting the user to choose which display surfaces to capture, the 
+            <a href="#dfn-user-agent">user agent</a> MUST allow
+            the user to choose any individual
+            <a data-cite="screen-capture#dfn-display-surface">display surface</a> type.
+            However, the <a href="#dfn-user-agent">user agent</a> MAY restrict
+            the choice so that it would not mix surfaces of a different type.
+          </p>
+          <p>
+            As with {{MediaDevices/getDisplayMedia()}}, {{PermissionState/"granted"}} permissions
+            cannot be persisted.
+          </p>
+          <p>When the {{MediaDevices/getDisplayMediaSet()}}
+            method is called, the user agent MUST run the following
+            steps:</p>
+          <ol>
+            <li>
+              <p>If the [=relevant global object=] of [=this=] does not have
+                [=transient activation=], return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{InvalidStateError}}.</p>
+            </li>
+            <li>
+              <p>Let <var>options</var> be the method's first
+                argument.</p>
+            </li>
+            <li>
+              <!-- Note: this is one difference to getDisplayMedia. -->
+              <p>
+                If <code>options</code> contains any attribute other than
+                <code>options.video</code>, return a promise <a>rejected</a>
+                with a {{DOMException}} object whose {{DOMException/name}}
+                attribute has the value {{TypeError}}.
+              </p>
+            </li>
+            <li>
+              <!-- Note: keeping the [...] notation to not diverge too far
+              from the getDisplayMediaSet spec. -->
+              <p>Let <var>constraints</var> be
+                <code>[<var>options</var>.<code>video</code><code>]</code>.
+              </p>
+            </li>
+            <li>
+              <p>If <code>constraints.video</code> is <code>false</code>,
+                return a promise [=reject|rejected=] with a newly
+                [=exception/created=] {{TypeError}}.</p>
```

Comment:
Or we could make gDMS accept no parameter. Do we expect to use any of the existing parameters? (Nobody has ever complained about an algorithm being too simple.)

Reply:
Parameters are gone now.